Software/app/basic_lorawan/conf/lorawan_conf.h

```diff
@@ -1,52 +1,87 @@
-  * @brief   Header for LoRaWAN middleware instances
-  ******************************************************************************
-  * @attention
-  *
-  * <h2><center>&copy; Copyright (c) 2020 STMicroelectronics.
-  * All rights reserved.</center></h2>
-  *
-  * This software component is licensed by ST under Ultimate Liberty license
-  * SLA0044, the "License"; You may not use this file except in compliance with
-  * the License. You may obtain a copy of the License at:
-  *                             www.st.com/SLA0044
-  *
-  ******************************************************************************
-  */
-
-/* Define to prevent recursive inclusion -------------------------------------*/
-#ifndef __LORAWAN_CONF_H__
-#define __LORAWAN_CONF_H__
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-#include "stm32_systime.h"
-
-/* Region configuration */
-/* 1- Configuration Region listed below will be linked in the MW code */
-/* 2- Must also configure ACTIVE_REGION in the application */
-
-#define REGION_EU868
-// #define REGION_US915
-
-#define HYBRID_ENABLED          0
-#define LORAMAC_CLASSB_ENABLED  0
-
-#ifndef CRITICAL_SECTION_BEGIN
-#define CRITICAL_SECTION_BEGIN( )      UTILS_ENTER_CRITICAL_SECTION( )
-#endif /* !CRITICAL_SECTION_BEGIN */
-#ifndef CRITICAL_SECTION_END
-#define CRITICAL_SECTION_END( )        UTILS_EXIT_CRITICAL_SECTION( )
-#endif /* !CRITICAL_SECTION_END */
-
-#ifdef __cplusplus
-}
-#endif
-
-#endif /* __LORAWAN_CONF_H__ */
-/************************ (C) COPYRIGHT STMicroelectronics *****END OF FILE****/
+/**
+  ******************************************************************************
+  * @file    lorawan_conf.h
+  * @author  MCD Application Team
+  * @brief   Header for LoRaWAN middleware instances
+  ******************************************************************************
+  * @attention
+  *
+  * <h2><center>&copy; Copyright (c) 2020 STMicroelectronics.
+  * All rights reserved.</center></h2>
+  *
+  * This software component is licensed by ST under BSD 3-Clause license,
+  * the "License"; You may not use this file except in compliance with the
+  * License. You may obtain a copy of the License at:
+  *                        opensource.org/licenses/BSD-3-Clause
+  *
+  ******************************************************************************
+  */
+
+/* Define to prevent recursive inclusion -------------------------------------*/
+#ifndef __LORAWAN_CONF_H__
+#define __LORAWAN_CONF_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "stm32_systime.h"
+#include "lora_info.h"
+
+/* Region ------------------------------------*/
+/* the region listed here will be linked in the MW code */
+/* the applic (on sys_conf.h) shall just configure one region at the time */
+/*#define REGION_AS923*/
+/*#define REGION_AU915*/
+/*#define REGION_CN470*/
+/*#define REGION_CN779*/
+/*#define REGION_EU433*/
+#define REGION_EU868
+/*#define REGION_KR920*/
+/*#define REGION_IN865*/
+#define REGION_US915
```

Comment:
@elsalahy When testing again I found something. It does still work, but I don't think that defining 2 regions is a good idea.

Reply:
Our node supports for a US915 and EU868 and the node can actually switch between frequencies.
This is a configuration for the middleware.

In lora_app.h the user defines `ACTIVE_REGION` to suit his needs.

Also, In the future it is ok to ask a question before requesting a change.